src/service/repository/repo.rs

```diff
@@ -78,7 +79,23 @@ impl Repository {
                 }
                 Ok(repo)
             }
-            (None, None) => Err(anyhow::anyhow!("No input provided")),
+            (None, None, Some(repo_url)) => {
+                let repo = repositories
+                    .filter(repo_url_col.eq(&repo_url))
+                    .load::<Repository>(connection)
+                    .map_err(|e| {
+                        error!("Error getting repository: {}", e);
+                        FailedToGetRepository(GetRepositoryError(e))
+                    })?;
+                if repo.is_empty() {
+                    return Err(anyhow::anyhow!(
+                        "Repository for repo url {} not found",
+                        &repo_url
+                    ));
+                }
+                Ok(repo)
+            }
+            (None, None, None) => Err(anyhow::anyhow!("No input provided")),
             _ => Err(anyhow::anyhow!("Invalid input")),
```

Comment:
```suggestion
            (Some(_), Some(_), None) | (Some(_), None, Some(_)) | (None, Some(_), Some(_)) => {
                Err(anyhow::anyhow!("Only one of id, user_id, or repo_url should be provided"))
            }
            (Some(_), Some(_), Some(_)) => {
                Err(anyhow::anyhow!("Cannot provide id, user_id, and repo_url simultaneously"))
            }
```

This change improves input validation by explicitly handling all possible input combinations and providing more descriptive error messages. It enhances the function's robustness, makes its behavior more predictable, and facilitates easier debugging by clearly communicating which input combinations are invalid and why.

Reply:
Nice. Added to last commit https://github.com/Extheoisah/hxckr-core/pull/17/commits/42d1b9f79e931dea9e862e025757782ae8906f32